src/Module/Markdown/Step/ProcessMarkdownFileName.php

```diff
@@ -13,8 +13,14 @@
  */
 class ProcessMarkdownFileName implements Step
 {
+
+    private $project;
```

Comment:
Services [should be stateless](https://en.wikipedia.org/wiki/Service_statelessness_principle), so it shouldn't store such data inside the class's state.

You should instead pass the `$project` variable from `__invoke()` to `renameReadme()` in a method parameter.